scapy/contrib/roce.py

```diff
@@ -179,8 +180,25 @@ def compute_icrc(self, p):
             pshdr[UDP].payload = Raw(bth + payload + icrc_placeholder)
             icrc = crc32(raw(pshdr)[:-4]) & 0xffffffff
             return self.pack_icrc(icrc)
+        elif isinstance(ip, IPv6):
+            # pseudo-LRH / IPv6 / UDP / BTH / payload
+            pshdr = Raw(b'\xff' * 8) / ip.copy()
+            pshdr.hlim = 0xff
+            pshdr.fl = 0xfffff
+            pshdr.tc = 0xff
+            pshdr[UDP].chksum = 0xffff
```

Comment:
The current implementation does not do it, but you should check for the existence of UDP and BTH headers before accessing them do make sure that Scapy won't crash if they don't. You can use the get method to do so.

Otherwise, this PR looks good to me. Thanks for your contribution!

Reply:
Thanks for the review!
UDP presence is checked in line 160
BTH - The crc_function is invoked post_build of the BTH header, so checking BTH presence is overkill.  
Thanks, Leonid

Reply:
I agree! I guess that is the limit of reviewing on a mobile phone.